hera_mc/mc_session.py

```diff
@@ -2307,26 +2307,588 @@ def add_correlator_config_file(self, config_hash, config_file):
 
         self.add(CorrelatorConfigFile.create(config_hash, config_file))
 
-    def get_correlator_config_file(self, config_hash):
+    def get_correlator_config_file(self, config_hash=None,
+                                   most_recent=None, starttime=None,
+                                   stoptime=None):
         """
         Get a correlator config file record from the M&C database.
 
+        If a config_hash is provided, the time-related optional keywords are ignored.
+
+        Default behavior is to return the most recent record(s) -- there can be
+        more than one if there are multiple records at the same time. If
+        starttime is set but stoptime is not, this method will return the first
+        record(s) after the starttime -- again there can be more than one if
+        there are multiple records at the same time. If you want a range of
+        times you need to set both startime and stoptime. If most_recent is set,
+        startime and stoptime are ignored.
+
         Parameters
         ----------
         config_hash : str
             Unique hash for config file.
+        most_recent : bool
+            If True, get most recent record. Defaults to True if starttime is
+            None.
+        starttime : astropy Time object
+            Time to look for records after. Ignored if most_recent is True,
+            required if most_recent is False.
+        stoptime : astropy Time object
+            Last time to get records for, only used if starttime is not None.
+            If none, only the first record after starttime will be returned.
+            Ignored if most_recent is True.
 
         Returns
         -------
-        list of CorrelatorConfigFile objects
+        list
+            List of CorrelatorConfigFile objects.
+        time_list : list of Astropy Time objects, optional
+            List of start times when the config files were used, same length as the list
+            of CorrelatorConfigFile objects. Only provided if `config_hash` is None.
 
         """
         from .correlator import CorrelatorConfigFile
```

Comment:
Is there a benefit to burying the import here instead of doing it at the top of the file? I generally prefer to do all imports at the top, unless there's a good reason not to (like it takes a long time or relies on optional modules).

Reply:
I don't think there is a real reason to do it in the method, but it is the pattern in all the other methods in this object. I'll play around with moving them up to the top.

Reply:
I moved all the imports from this package and from standard lib or our requirements to the top.